warp10/src/main/java/io/warp10/script/WarpScriptTraceableStatement.java

```diff
@@ -0,0 +1,116 @@
+//
+//   Copyright 2023  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script;
+
+/**
+ * Stores a statement, a type, and its position.
+ * This class is json serializable, snapshotable and, is a WarpScriptStackFunction.
+ * Type is defined by {@link WarpScriptTraceableStatement.STATEMENT_TYPE}
+ */
+public class WarpScriptTraceableStatement implements WarpScriptStackFunction {
```

Comment:
The name of the class should reflect its actual utility, this is not a Traceable statement but rather an Audit statement, so name should not induce confusion.